### PR TITLE
Guide external contributors on creating their first PR.

### PR DESCRIPTION
**Pull request by carlosonunez**
**What this PR does / why we need it**:

This pull request includes documentation to help new contributors
configure their test environment, navigate their way around our tests
and codebase, and use the `docker` infrastructure provider to
run exploratory tests locally.

**Which issue(s) this PR fixes**:

Fixes #

**Describe testing done for PR**:

I tested it locally with a Markdown viewer.

**Special notes for your reviewer**:

**Release note**:

```release-note

Adds a guide for new contributors to contribute to tanzu-framework

```
**New PR Checklist**

- [X] Ensure PR contains only public links or terms
- [X] Use good commit [messages](https://github.com/vmware-tanzu/tanzu-framework/blob/main/CONTRIBUTING.md)
- [X] Squash the commits in this branch before merge to preserve our git history
- [X] If this PR is just an idea or POC, use a [Draft PR](https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/about-pull-requests#draft-pull-requests) instead of a full PR
- [X] Add appropriate [kind label](../docs/release/kind-labels.md) according to what type of issue is being addressed.

### CODE REVIEW

File: docs/dev/your_first_pr.md [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+Thanks for choosing to help improve the Tanzu Framework! This guide will
+help you build your first successful pull request.
+
+# Opening an Issue
+
+Your new idea starts with a new GitHub issue. You'll link your pull request to
+your issue once you're ready to submit it. This will help us appropriately
+prioritize the release into which your PR should get merged into.
+
+[Click here](https://github.com/vmware-tanzu/tanzu-framework/issues/new) to
```
Comment (stmcginnis) [inline docs/dev/your_first_pr.md:12]: Could we get this updated to use `https://github.com/vmware-tanzu/tanzu-framework/issues/new/choose` instead so it goes through our templates?

File: docs/dev/your_first_pr.md [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+open your first issue.
+
+# Getting the Code
+
+Now that you've created your issue, your next step is to fork the repo.
+
+Your pull request will merge your fork into our repository.
+
+Click on the "Fork" button at the top right of the page to fork
+Tanzu Framework into your Github account.
```
Comment (stmcginnis) [inline docs/dev/your_first_pr.md:22]: nitpick in case you update, please use proper casing of "GitHub" :)

File: docs/dev/your_first_pr.md [hunk 3]
```diff
@@ -0,0 +1,251 @@
...
+# Preparing Your Development Environment
+
+Next, we'll need to prepare your development environment.
+
+Before doing anything, go ahead and clone your fork with `git clone`.
+
+## Download and install Golang and GVM
+
+Tanzu Kubernetes Grid is written in [Golang](https://golang.org), or Go
+for short. While it is currently built against Go v1.1.7, this might change
```
Comment (stmcginnis) [inline docs/dev/your_first_pr.md:35]: Transposed that Go version. I think we're using go v1.17 right now. Or even still v1.16.

File: docs/dev/your_first_pr.md [hunk 4]
```diff
@@ -0,0 +1,251 @@
...
+```
+
+Next, we'll build the CLI components. The `manager` binary is used as an entrypoint
+for various TKG components.
+
+# Create a test!
+
+## Ginkgo and Gomega
+
+TKG uses [Ginkgo](https://onsi.github.io/ginkgo/) to create tests in
```
Comment (stmcginnis) [inline docs/dev/your_first_pr.md:99]: Should maybe refer to "Tanzu framework" here instead of "TKG".

File: docs/dev/your_first_pr.md [hunk 5]
```diff
@@ -0,0 +1,251 @@
...
+[Gherkin](https://cucumber.io/docs/gherkin/) syntax. Gherkin is a syntax
+used for enabling [behavior-driven development](https://en.wikipedia.org/wiki/Behavior-driven_development).
+Matchers are Ginkgo are provided by [Gomega](https://onsi.github.io/gomega).
```
Comment (stmcginnis) [inline docs/dev/your_first_pr.md:102]: s/Matchers are/Matchers in/ ?

**Review by stmcginnis — COMMENTED** [review]
Great stuff! Several minor things pointed out inline, but nothing too major.

**Review by carlosonunez — COMMENTED** [review]
@stmcginnis PTAL

**Comment by carlosonunez** [comment]
@stmcginnis I updated this PR to match the newest iteration of our `CONTRIBUTING` doc. Can you PTAL? Thanks!

**Review by stmcginnis — APPROVED** [review]
Looks great!

I've kicked off the failing job again to see if that was just a transient failure. It shouldn't be caused by this PR, so if it still fails we may need to rebase this on `main` to pick up any fixes.

**Comment by carlosonunez** [comment]
Thanks! I’ll give it another rebase just in case.

**Comment by randomvariable** [comment]
This is unfortunately outdated by now. Sorry.

**Comment by carlosonunez** [comment]
all good!-- CarlosOn Nov 25, 2022, at 05:12, Naadir Jeewa ***@***.***> wrote:﻿
Closed #798.

**Pull request closed without merging**